src/models/node.js

```diff
@@ -900,21 +901,25 @@ const Node = {
       const previous = this.getPreviousText(startKey)
       if (!previous || !previous.length) return []
       const char = previous.characters.get(previous.length - 1)
-      return char.marks.toArray()
+      return options.perCharacter ? [char.marks.toArray()] : char.marks.toArray()
```

Comment:
Would this warrant it's own subset of functions?  `getMarksPerCharacter` and `getMarksPerCharacterAsArray`?